### PR TITLE
Overlay

### PR DESCRIPTION
**Pull request by eonpatapon**
```
$ nix-instantiate -A contrail32.version --eval
"3.2"
$ nix-instantiate -A contrail32.sources.generateds.rev --eval
"706ac48789084ff3148c250e102842df46dae383"
$ nix-instantiate -A contrail34.version --eval
"3.4"
$ nix-instantiate -A contrail34.sources.generateds.rev --eval
"706ac48789084ff3148c250e102842df46dae383"
$ nix-instantiate -A contrail41.sources.generateds.rev --eval
"836c6d3fd0bfaf0b6c832db8827a45b14053c619"
```

### CODE REVIEW

File: contrail-overlay.nix [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    callPackage = callPackageWith (self // { contrail = lself; });
+    callPackages = callPackagesWith (self // { contrail = lself; });
+
+  in {
+
+    sources = null;
+    version = null;
+
+    # this is used to do contrail.override then next overrides will use override'
+    override = f:
```
Comment (nlewo) [inline contrail-overlay.nix]: `override = override' contrailBuild` would be simpler.

Comment (eonpatapon) [reply contrail-overlay.nix]: done

**Comment by eonpatapon** [comment]
```
$ git checkout 008ca1e1b8a237a581100170c35ca3657a2ffd80
$ nix-instantiate -A contrail32 | sha256sum
22eab7455e7a790f026997afc6608b423090b044c37a641365e3937032957899

$ git checkout overlay
$ nix-instantiate -A contrail32 | grep -v -E '(controller|contrail-python|third-party|workspace)' | sha256sum
22eab7455e7a790f026997afc6608b423090b044c37a641365e3937032957899
```

**Comment by eonpatapon** [comment]
$ git checkout 008ca1e1b8a237a581100170c35ca3657a2ffd80
$ nix-instantiate jobset.nix -A contrail32.pythonPackages  | sha256sum
ea186dbf3342d00ac1321a77f9fae09be45e01109647ad328fc4ace9bba81b05

$ git checkout overlay
$ nix-instantiate jobset.nix -A contrail32.pythonPackages | grep -v -E '(vrouter|discovery|sandesh|openstack)' | sha256sum
ea186dbf3342d00ac1321a77f9fae09be45e01109647ad328fc4ace9bba81b05

**Comment by eonpatapon** [comment]
test.allInOne sha changes because it was using pkgs from <nixpkgs>.

**Comment by eonpatapon** [comment]
```
$ nix-instantiate -A contrail32 | grep -v -E '(controller|contrail-python|third-party|workspace|cassandra-cpp-driver|libgrok)' | sha256sum
22eab7455e7a790f026997afc6608b423090b044c37a641365e3937032957899
```

**Comment by eonpatapon** [comment]
test.allInOne sha changes because in the test we had:

```
_module.args = { inherit contrailPkgs isContrail32 isContrailMaster; };
```

So it was not using `pkgs` from `nixpkgs-fetch.nix`

**Pull request merged**